### DISCUSSION TITLE
Fibaro FGRGBW Controller targetColor

### DISCUSSION THREAD

**Discussion by hestiap** [zwave-js/zwave-js-ui#1989] [Q&A] [answered]
Hi,
I’d like to change the color of this RGBW controller via mqtt.
Via the zwavejsmqtt GIU, I could do
 
![image](https://user-images.githubusercontent.com/73014953/143106524-6bcb4e4d-7c70-47a4-8c3c-aa9da79a4647.png)

Which gave in mqtt
`{"time":1637699047658,"value":{"warmWhite":0,"red":111,"green":0,"blue":0},"nodeName":"LED Terrasse","nodeLocation":"Jardin"}
`
I’d tried to do the same via dz with no success
`mosquitto_pub -t zwave/95/51/0/targetColor -m {\"value\":\"warmWhite\":75,\"red\":0,\"green\":13,\"blue\":100}`
and in mqtt
`{"value":"warmWhite":75,"red":0,"green":13,"blue":100}`
But nothing on zwavejsmqtt log ☹
Any help will be welcome

**Comment by robertsLando** [comment]
Check docs, to set values you have to add `/set` suffix:

`mosquitto_pub -t zwave/95/51/0/targetColor/set -m {\"value\":\"warmWhite\":75,\"red\":0,\"green\":13,\"blue\":100}`

**Reply by hestiap** [reply]
Thanks,
I wasn't sure it was a "set"

I had tried it also with no success, but perhaps with other mistakes
I've done it again with your line!
out of dz:
`mosquitto_pub -t zwave/95/51/0/targetColor/set -m {\"value\":\"warmWhite\":75,\"red\":0,\"green\":13,\"blue\":100}`

in mqtt:
`{"value":"warmWhite":75,"red":0,"green":13,"blue":100}`

in zwavejs2mqtt: 
`2021-11-24 08:34:46.760 INFO MQTT: Message received on zwave/95/51/0/targetColor/set: '{"value":"warmWhite":75,"red":0,"green":13,"blue":100}'
2021-11-24 08:34:46.763 INFO ZWAVE: Writing '{"value":"warmWhite":75,"red":0,"green":13,"blue":100}' to 95-51-0-targetColor
2021-11-24 08:34:46.766 ERROR ZWAVE: Unable to write '{"value":"warmWhite":75,"red":0,"green":13,"blue":100}' on 95-51-0-targetColor`

`Driver: Color Switch: "targetColor" must be set to an object which specifies each color channel (ZW0322)`

So it didn't work neither as is

**Reply by robertsLando** [reply]
You are missing a `{`.

`mosquitto_pub -t zwave/95/51/0/targetColor/set -m {\"value\":{\"warmWhite\":75,\"red\":0,\"green\":13,\"blue\":100}}`

**Comment by hestiap** [comment, marked as answer]
Thanks again,
it works
